polling/testservice.go

```diff
@@ -0,0 +1,26 @@
+package polling
+
+import (
+	"context"
+	"errors"
+)
+
+type TestService struct {
+	tryAttempts   int
+	errorAttempts int
+}
+
+func NewTestService(errorAttempts int) TestService {
+	return TestService{
+		tryAttempts:   0,
+		errorAttempts: errorAttempts,
+	}
+}
+
+func (testService *TestService) GetServiceResult(ctx context.Context) (string, error) {
```

Comment:
Mentioned above, but I would differentiate poll fail from service fail for clarity. Not required but would be clearer.

Reply:
How is poll failing possible here? I am not sure I understood